src/themes/default/components/core/blocks/Header/MicrocartIcon.vue

```diff
@@ -25,11 +25,7 @@ import MicrocartIcon from '@vue-storefront/core/compatibility/components/blocks/
 export default {
   // mixins: [MicrocartIcon],
   mounted () {
-    document.addEventListener('visibilitychange', () => {
-      if (!document.hidden) {
-        this.$store.dispatch('cart/load')
-      }
-    })
+    window.addEventListener('storage', this.getItemsFromStorage)
```

Comment:
wdyt about moving all of this logic to mounted? It's easier to read code (everything is in one place) and we don't need to add method to component
```
    const getItemsFromStorage =...
    window.addEventListener('storage', getItemsFromStorage)
    this.$once('hook:destroyed', () => {
      window.removeEventListener('storage', getItemsFromStorage)
    })
```

Reply:
I think `beforeDestroy` it's even dedicated place to removing event listeners.

Reply:
I moved the functions to a separate file. Thanks for the suggestions!